app/main.py

```diff
@@ -1,2 +1,118 @@
+        self.size = 0
+        self.hash_table: list[Dictionary.Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable, hash_value: int) -> int:
+        index = hash_value % self.capacity
+
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+        return index
+
+    @property
+    def current_max_size(self) -> int:
+        return int(self.capacity * self.resize_threshold)
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= self.capacity_multyplier
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+
+        for node in old_hash_table:
+            if node is not None:
+                self[node.key] = node.value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        hash_value = hash(key)
+        index = self._calculate_index(key, hash_value)
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                self.hash_table[index].value = value
+                return
+            index = (index + 1) % self.current_max_size
+
+        self.hash_table[index] = Dictionary.Node(key, value, hash_value)
+        self.size += 1
+
+        if self.size >= self.current_max_size:
+            self.resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_value = hash(key)
+        index = self._calculate_index(key, hash_value)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        return self.hash_table[index].value
+
+    def __delitem__(self, key: Hashable) -> None:
+        hash_value = hash(key)
+        index = self._calculate_index(key, hash_value)
+
+        while True:
+            if self.hash_table[index] is None:
+                raise KeyError(f"Cannot find value for key: {key}")
+            elif self.hash_table[index].key == key:
+                break
+            index = (index + 1) % self.current_max_size
+
+        self.hash_table[index] = None
+        self.size -= 1
+
+        next_index = (index + 1) % self.current_max_size
+        while self.hash_table[next_index] is not None:
+            node_to_rehash = self.hash_table[next_index]
+            self.hash_table[next_index] = None
+            self.size -= 1
+            self.__setitem__(node_to_rehash.key, node_to_rehash.value)
+            next_index = (next_index + 1) % self.current_max_size
+
+    def get(self, key: Hashable, default: Optional[Any] = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __str__(self) -> str:
+        return str(self.hash_table)
+
+    def clear(self) -> None:
+        self.__init__()
```

Comment:
If you call init it will reduce your list capacity to original - 8 cells which is not very accurate. __init__ is usually called once on object creation, and it is not recommended to call it many times